sdk/openai/openai/samples/cookbook/speechToSpeechChat/src/index.js

```diff
@@ -0,0 +1,122 @@
+  speechKey = document.getElementById("speechKey").value;
+  speechRegion = document.getElementById("speechRegion").value;
+
+  const speechConfig = SpeechSDK.SpeechConfig.fromSubscription(speechKey, speechRegion);
+  const audioConfig = SpeechSDK.AudioConfig.fromDefaultMicrophoneInput();
+  recognizer = new SpeechSDK.SpeechRecognizer(speechConfig, audioConfig);
+
+  statusDiv = document.getElementById("statusDiv");
+  const promptInput = document.getElementById("promptInput");
+  statusDiv.innerHTML = "";
+  promptInput.innerHTML = "";
+
+  recognizer.recognized = (s, e) => {
+    if (!!e && !!e.result && !!e.result.text) {
+      statusDiv.innerHTML = "Sending input prompt to ChatGPT";
+      promptInput.innerHTML = e.result.text;
+      
+      sendTextToChatGPT(e.result.text);
+    }
+  };
+
+  recognizer.canceled = (s, e) => {
+    console.log(e);
+    statusDiv.innerHTML = "(Recognition Canceled) Reason: " + SpeechSDK.CancellationReason[e.reason];
+    if (e.reason === SpeechSDK.CancellationReason.Error) {
+        statusDiv.innerHTML += ": " + e.errorDetails;
+    }
+    statusDiv.innerHTML += "\r\n";
+    recognizer.close();
+    recognizer = undefined;
+  };
+
+  try {
+    statusDiv.innerHTML = "Listening from microphone input for ChatGPT prompt";
+    recognizer.startContinuousRecognitionAsync(
+      () => {},
+      function (err) {
+        window.console.log(err);
+        recognizer.close();
+        recognizer = undefined;
+      });
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function sendTextToChatGPT(text) {
+  if (!client) {
+    const deploymentId = document.getElementById("deploymentId");
+    const azureKey = document.getElementById("azureKey");
+    const endpoint = document.getElementById("endpoint");
+
+    const credential = new AzureKeyCredential(azureKey.value);
+    client = new OpenAIClient(endpoint.value, credential);
+  }
+
+  async function showResponseChoices() {
+    try {
+      const { choices } = await client.getCompletions(deploymentId.value, [text]);
+      const gptResponseText = choices[0]?.text;
+      speakText(gptResponseText);
+      const resultDiv = document.getElementById("resultDiv");
+      resultDiv.innerHTML = gptResponseText;
+    } catch (e) { 
+      console.log(e);
+    }
+  }
+  showResponseChoices();
+}
+
+function speakText(text) {
+  const speechConfig = SpeechSDK.SpeechConfig.fromSubscription(speechKey, speechRegion);
+  const audioConfig = SpeechSDK.AudioConfig.fromDefaultSpeakerOutput();
+  let synthesizer = new SpeechSDK.SpeechSynthesizer(speechConfig, audioConfig);
+  synthesizer.speakTextAsync(
+    text,
+    function (result) {
+      if (result.reason === SpeechSDK.ResultReason.SynthesizingAudioCompleted) {
+        statusDiv.innerHTML += "\n\nsynthesis finished for [" + text + "].\n";
+      } else if (result.reason === SpeechSDK.ResultReason.Canceled) {
+        statusDiv.innerHTML += "\n\nsynthesis failed. Error detail: " + result.errorDetails + "\n";
```

Comment:
We can probably use [Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) when possible for these strings